main.cpp

```diff
@@ -40,13 +36,23 @@ int main(int argc, char* argv[]) {
 			return 1;
 		}
 
-
-		// Run the lexer.
-		while (not tok.eof()) {
-			tok = lir::lexer::run(files);
-			// lir::println("[L] -> ", tok);  // Print tokens (don't use this for big files.)
-		}
-
+		lir::AST ast = lir::parser::run(files);
+		//lir::println("[T] -> ", ast);
+		std::cout << ast << std::endl;
+
+		auto result = lir::backend::treewalk::interpret(ast);
+
+		std::cout << "[R] -> "
```

Comment:
Maybe add a debug flag to enable printing stuff, same for the `std::cout << ast` above.